actions/emoji_actions.js

```diff
@@ -79,15 +78,26 @@ export function addRecentEmoji(alias) {
         }
 
         let updatedRecentEmojis;
-        if (recentEmojis.includes(name)) {
-            // If the emoji is already in the recent list, remove it and add it to the front
-            const recentEmojisFiltered = recentEmojis.filter((recentEmoji) => recentEmoji !== name);
-            updatedRecentEmojis = [...recentEmojisFiltered, name].slice(-MAXIMUM_RECENT_EMOJI);
+        const currentEmojiIndexInRecentList = recentEmojis.findIndex((recentEmoji) => recentEmoji.name === name);
+        if (currentEmojiIndexInRecentList > -1) {
+            const currentEmojiInRecentList = recentEmojis[currentEmojiIndexInRecentList];
+
+            // If the emoji is already in the recent list, remove it and add it to the front with updated usage count
+            const updatedCurrentEmojiData = {
+                name,
+                usageCount: currentEmojiInRecentList.usageCount + 1,
+            };
+            recentEmojis.splice(currentEmojiIndexInRecentList, 1);
+            updatedRecentEmojis = [...recentEmojis, updatedCurrentEmojiData].slice(-MAXIMUM_RECENT_EMOJI);
         } else {
-            updatedRecentEmojis = [...recentEmojis, name].slice(-MAXIMUM_RECENT_EMOJI);
+            const currentEmojiData = {
+                name,
+                usageCount: 1,
+            };
+            updatedRecentEmojis = [...recentEmojis, currentEmojiData].slice(-MAXIMUM_RECENT_EMOJI);
         }
 
-        LocalStorageStore.setRecentEmojis(currentUserId, updatedRecentEmojis);
+        _dispatch(savePreferences(currentUserId, [{category: Constants.Preferences.RECENT_EMOJIS, name: currentUserId, user_id: currentUserId, value: JSON.stringify(updatedRecentEmojis)}]));
```

Comment:
Could you rename `_dispatch` back to `dispatch` since it's no longer unused?

@M-ZubairAhmed I think you had changed this a few weeks back when we stopped using dispatch in this function. I think we should probably keep the existing name in that situation since we'll end up having to just change it back later